cadc-storage-permissions/src/main/java/org/opencadc/inventory/permissions/PermissionsClient.java

```diff
@@ -84,38 +92,155 @@
+        InventoryUtil.assertNotNull(
+            PermissionsClient.class, "serviceID", serviceID);
+        this.serviceURL = getServiceURL(serviceID);
     }
     
     /**
-     * Get the read permissions information about the file identified by fileURI.
-     * 
-     * @param artifactURI Identifies the artifact for which to retrieve grant information.
+     * Get the read permissions information about the file
+     * identified by artifactURI.
+     *
+     * @param artifactURI Identifies the artifact for which to retrieve grant
+     *                    information.
      * @return The read grant information.
      * 
      * @throws ResourceNotFoundException If the file could not be found.
-     * @throws TransientException If an unexpected, temporary exception occurred. 
+     * @throws TransientException If an unexpected,
+     *                            temporary exception occurred.
      */
     public ReadGrant getReadGrant(URI artifactURI)
         throws ResourceNotFoundException, TransientException {
-        return null;
+
+        return (ReadGrant) getGrant(artifactURI, Operation.read);
     }
 
     /**
-     * Get the write permissions information about the file identified by fileURI.
+     * Get the write permissions information about the file
+     * identified by artifactURI.
      *
-     * @param artifactURI Identifies the artifact for which to retrieve grant information.
+     * @param artifactURI Identifies the artifact for which to
+     *                    retrieve grant information.
      * @return The write grant information.
      *
      * @throws ResourceNotFoundException If the file could not be found.
-     * @throws TransientException If an unexpected, temporary exception occurred.
+     * @throws TransientException If an unexpected,
+     *                            temporary exception occurred.
      */
     public WriteGrant getWriteGrant(URI artifactURI)
         throws ResourceNotFoundException, TransientException {
-        return null;
+
+        return (WriteGrant) getGrant(artifactURI, Operation.write);
+    }
+
+    /**
+     * Get the permission information about the file identified by artifactURI.
+     *
+     * @param artifactURI Identifies the artifact for which to
+     *                    retrieve grant information.
+     * @param op The type of grant to retrieve.
+     * @return The grant information.
+     * @throws ResourceNotFoundException If the file could not be found.
+     * @throws TransientException If an unexpected,
+     *                            temporary exception occurred.
+     */
+    Grant getGrant(URI artifactURI, Operation op)
+        throws ResourceNotFoundException, TransientException {
+
+        URL grantURL = getGrantURL(serviceURL, op, artifactURI);
+
+        // TODO is the client doing retries???
+        OutputStream out = getOutputStream();
+        HttpGet httpGet = getHttpGet(grantURL, out);
+        httpGet.run();
```

Comment:
Main review issue: When using HtpGet, you are better off calling prepare(), deciding which exceptions to let fly and which to catch (let fly is preferrable), and if there are no exceptions then call getInputStream() to read. That way you don't have to reproduce the error handling at all and should not have to check status codes.